XenModel/Alerts/PolicyAlert.cs

```diff
@@ -30,29 +30,67 @@
  */
 
 using System;
+using System.Text.RegularExpressions;
+using System.Collections.Generic;
 using System.Globalization;
 using System.Text;
 using System.Xml;
 using XenAdmin.Network;
 using XenAPI;
 
+
 namespace XenAdmin.Alerts
 {
     public class PolicyAlert : IEquatable<PolicyAlert>
     {
         public readonly string Type;
         public readonly string Text;
         public readonly DateTime Time;
+        public readonly string PolicyType;
+        public readonly int numberOfVMsFailed;
 
-        public PolicyAlert(long priority, string name, DateTime _time)
+        public PolicyAlert(long priority, string name, DateTime _time, string body)
         {
             Type = (priority == 4 ? "info": "error");
             Time = _time;
-            Text = Message.FriendlyBody(name);
+            PolicyType = "VMSS";
+
+            if(Type == "info")
+            {
+                Text = Message.FriendlyBody(body);
+                return;
+            }
+
+            /* We reach here when message type is an error hence we need 
+             * to parse the message body accordingly */
+
+            numberOfVMsFailed = Regex.Matches(body, "VM:").Count;
+            string[] messageHeader = Regex.Split(body.Replace("\n", " "), "Details:");
+            Text = messageHeader[0] + "\n\nDetails:\n";

```

Comment:
Here the Text won't be localizable, because messageHeader[0] is the taken straight from Message.body and "Details" is hardcoded.
I suggest to use Message.FriendlyBody(name) (which will be different for each VMSS message) in a string.Format() call.

Also, I would like to include the policy name in here, which I think we can obtain from the Message object.

So, for example if `Message.body-vmss_snapshot_failed` is something like: _"The snapshot schedule '{0}' failed because a snapshot could not be created on {1} VMs. \n {2}"_, then it will be used like:
`string.Format(Message.FriendlyBody(name), policyName, nmberOfVMsFailed, details);`, where details will contain an entry for each VM that encountered an error.

I think this will work with the other VMSS messages (VMSS_SNAPSHOT_MISSED_EVENT, VMSS_XAPI_LOGON_FAILURE, etc. - if any of them is considered "error") even if the message body does not contain detailed error information.